backend/src/main/java/com/redhat/cloud/notifications/db/repositories/BehaviorGroupRepository.java

```diff
@@ -395,6 +395,76 @@ public void updateEventTypeBehaviors(String orgId, UUID eventTypeId, Set<UUID> b
         }
     }
 
+    /**
+     * Deletes the specified behavior group from the event type, by deleting the relation in the "event_type_behavior"
+     * table.
+     * @param eventTypeUuid the event type to remove the behavior group from.
+     * @param behaviorGroupUuid the behavior group to remove.
+     * @param orgId the org id to make sure that the caller has visibility on that behavior group, so that they don't
+     *              delete unowned relations.
+     */
+    @Transactional
+    public void deleteBehaviorGroupFromEventType(final UUID eventTypeUuid, final UUID behaviorGroupUuid, final String orgId) {
+        // First check that the specified event type exists.
+        final var eventTypeExistsQuery =
+                "SELECT " +
+                    "1 " +
+                "FROM " +
+                    "EventType AS e " +
+                "WHERE " +
+                    "e.id = :eventTypeUuid";
+
+        try {
+            this.entityManager.createQuery(eventTypeExistsQuery)
+                .setParameter("eventTypeUuid", eventTypeUuid)
+                .getSingleResult();
+        } catch (final NoResultException e) {
+            throw new BadRequestException("the specified event type does not exist");
+        }
+
+        // Second, check that the behavior group also exists.
+        final var behaviorGroupExistsQuery =
+                "SELECT " +
+                        "1 " +
+                "FROM " +
+                    "BehaviorGroup AS bg " +
+                "WHERE " +
+                    "bg.id = :behaviorGroupUuid " +
+                "AND " +
+                    "bg.orgId = :orgId";
+
+        try {
+            this.entityManager.createQuery(behaviorGroupExistsQuery)
+                .setParameter("behaviorGroupUuid", behaviorGroupUuid)
+                .setParameter("orgId", orgId)
+                .getSingleResult();
+        } catch (final NoResultException e) {
+            throw new BadRequestException("the specified behavior group does not exist");
+        }
```

Comment:
I'm not sure we need to run these checks (which put pressure on the DB). If the event type or the behavior group are not found, then the following query will be executed and will not delete anything. The client will also be informed that no records were found with the event type / behavior group couple.

Reply:
All right! I removed those checks!